pkg/provider/ecs/ecs.go

```diff
@@ -408,21 +379,21 @@ func (p *Provider) listInstances(ctx context.Context, client *awsClient) ([]ecsI
 	return instances, nil
 }
 
-func (p *Provider) lookupMiInstances(ctx context.Context, client *awsClient, clusterName *string, ecsDatas map[string]*ecs.Task) (map[string]*ssm.InstanceInformation, error) {
+func (p *Provider) lookupMiInstances(ctx context.Context, client *awsClient, clusterName *string, ecsDatas map[string]ecstypes.Task) (map[string]ssmtypes.InstanceInformation, error) {
 	instanceIDs := make(map[string]string)
-	miInstances := make(map[string]*ssm.InstanceInformation)
+	miInstances := make(map[string]ssmtypes.InstanceInformation)
 
-	var containerInstancesArns []*string
-	var instanceArns []*string
+	var containerInstancesArns []string
+	var instanceArns []string
```

Comment:
AWS SDK V2 only uses pointers for input parameters when necessary. See https://aws.github.io/aws-sdk-go-v2/docs/migrating/#pointer-parameters.

The same applies to other areas in this file.